### PR TITLE
SHIELD-17 doc first draft for pg upgrade

### PR DESCRIPTION
**Pull request by vivekshankar1**
Signed-off-by: Vivek Shankar <vshankar@progress.com>

### :nut_and_bolt: Description: What code changed, and why?

Contains the docs for major upgrade and Postgres migration along with clear data command.
It includes all scenarios like normal and airgapped for both external and embedded postgresql

### :chains: Related Resources

https://chefio.atlassian.net/browse/SHIELD-17?atlOrigin=eyJpIjoiNzA3YzRkY2U2MjM5NDBjMWEyNDhjZDc3MzI2NWMyM2YiLCJwIjoiaiJ9

### :+1: Definition of Done

Docs steps should match the commands and user flow

### :athletic_shoe: How to Build and Test the Change

check the netlify link

### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/docs-chef-io/content/automate/major_upgrade.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+## Upgrade Journey
+
+Please choose following upgrade journey based on your current version of chef automate.All these upgrades are mannual upgrades.
+
+| Your Current Version | Upgrade To |
+| -------------------- | ---------- |
+| Any version before 2022012400000| 2022012400000|
+| on 2022012400000| 3.0.1|
+| 3.0.1| 4.0.1|
```
Comment (vivekshankar1) [inline components/docs-chef-io/content/automate/major_upgrade.md:25]:
```suggestion
```
Not required as this upgrade talks about upgrade till version 3 and not beyond that

File: components/docs-chef-io/content/automate/major_upgrade.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+For example, if today you are on version 2021201164433. Then your Upgrade Journey should be:<br />
+1.Mannual upgrade to 2022012400000.<br />
+2.Then mannual upgrade to 3.0.1.<br />
+3.Then mannual upgrade to 4.0.1.
```
Comment (vivekshankar1) [inline components/docs-chef-io/content/automate/major_upgrade.md:30]:
```suggestion
```
Not required as this upgrade talks about upgrade till version 3 and not beyond that

File: components/docs-chef-io/content/automate/major_upgrade.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+* Copy the CLI (chef-automate) and AIB (automate_v22.0.1.aib) to Air Gapped Machine where you are running the Chef Automate.
+* Run **Upgrade** in Air Gapped Machine using `sudo chef-automate upgrade run --airgap-bundle /<path>/automate_v3.x.x.aib --major` command. Once the Chef Automate gets upgraded you will get:
+  * List of upgraded information.
+  * Checklist of requirements pre-upgrade.
+  * List of steps to perform post-upgrade.
+
+* Check the status of the version upgrade using `$ chef-automate upgrade status` command.
+* Migrate your data from PostgreSQL **v9.6** to **v13** by running `chef-automate post-major-upgrade migrate --data=PG` command.
+* Check whether all the services are running using `chef-automate status` command.
+* Rectify if all the data is present in your upgraded Chef Automate version. If yes, clear the old PostgreSQL data using `chef-automate post-major-upgrade clear-data --data=PG` command.
```
Comment (IanMadd) [inline components/docs-chef-io/content/automate/major_upgrade.md:104]: > Rectify if all the data is present in your upgraded Chef Automate version.

How do they do that?

File: components/docs-chef-io/content/automate/major_upgrade.md [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+| 20220329091442| 3.0.1|
+
+For example, if today you are on version 2021201164433, then your upgrade journey should be:
+
+1. Manual upgrade to 20220329091442.
+1. Manual upgrade to 3.0.1.
+
+## Prerequisites
+
+- **Plan your downtime**: This upgrade requires downtime. Before upgrading, set the environment to handle the downtime.
```
Comment (IanMadd) [inline components/docs-chef-io/content/automate/major_upgrade.md:32]: > Before upgrading, set the environment to handle the downtime.
How do users do that? Do we have docs for this somewhere?

File: components/docs-chef-io/content/automate/major_upgrade.md [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+   chef-automate post-major-upgrade migrate --data=PG
+   ```
+
+1. Verify that all services are running:
+
+   ```sh
+   chef-automate status
+   ```
+
+1. Verify that all the data is present in your upgraded Chef Automate version. If yes, clear the old PostgreSQL data:
```
Comment (IanMadd) [inline components/docs-chef-io/content/automate/major_upgrade.md:108]: > Verify that all the data is present in your upgraded Chef Automate version.

How should users do that? Are there any commands they can run or other checks?

**Comment by vivekshankar1** [comment]
Will the upgrade to 3.x subsection be moved under a common upgrade menu ?

File: components/docs-chef-io/content/automate/major_upgrade.md [hunk 6]
```diff
@@ -0,0 +1,340 @@
++++
+title = "Upgrade Chef Automate to 3.X.X"
```
Comment (vivekshankar1) [inline components/docs-chef-io/content/automate/major_upgrade.md:2]:
```suggestion
title = "Upgrade Chef Automate to 3.x"
```

File: components/docs-chef-io/content/automate/major_upgrade.md [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+date = 2022-03-03T12:02:46-08:00
+draft = false
+gh_repo = "automate"
+
+[menu]
+  [menu.automate]
+    title = "Upgrade to 3.X.X"
```
Comment (vivekshankar1) [inline components/docs-chef-io/content/automate/major_upgrade.md:9]:
```suggestion
    title = "Upgrade to 3..x"
```

**Comment by IanMadd** [comment]
> Will the upgrade to 3.x subsection be moved under a common upgrade menu ?

Yeah, that's a good idea.

File: components/docs-chef-io/content/automate/major_upgrade.md [hunk 8]
```diff
@@ -0,0 +1,345 @@
...
+Chef Automate provides an entire suite of enterprise capabilities for node visibility and compliance. Chef Automate upgrades from one minor version to another automatically. However, Chef Automatate will not automatically upgrade to a major version. See the instructions below for manually upgrading Chef Automate from date-based versions to Chef Automate 3.
+
+## Upgrade Journey
+
+Please choose following upgrade journey based on your current version of Chef Automate. All of these upgrades are manual upgrades.
+
+| Your Current Version | Upgrade To |
+| -------------------- | ---------- |
+| Any version before 20220329091442| 20220329091442|
+| 20220329091442| 3.0.1|
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/major_upgrade.md:24]:
```suggestion
| 20220329091442| 3.0.x|
```

File: components/docs-chef-io/content/automate/major_upgrade.md [hunk 9]
```diff
@@ -0,0 +1,345 @@
...
+
+For example, if today you are on version 2021201164433, then your upgrade journey should be:
+
+1. Manual upgrade to 20220329091442.
+1. Manual upgrade to 3.0.1.
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/major_upgrade.md:29]:
```suggestion
1. Manual upgrade to 3.0.x
```

File: components/docs-chef-io/content/automate/major_upgrade.md [hunk 10]
```diff
@@ -0,0 +1,345 @@
...
+
+{{< /warning >}}
+
+### Chef Automate With Embedded PostgreSQL
+
+To upgrade Chef Automate with embedded PostgreSQL, follow the steps given below:
+
+**Upgrade Chef Automate**
+
+1. If you haven't already done so, upgrade to the last date-based version of Chef Automate:
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/major_upgrade.md:63]:
```suggestion
1. Your current version should be 20220329091442. To check the version run:
\```sh
sudo chef-automate version
\```
If you are on an earlier version, then upgrade to latest version of date series, by running:
```

**Comment by dishanktiwari2501** [comment]
Upgraded the TOC in [3802](https://github.com/chef/chef-web-docs/pull/3802).

**Pull request merged**